src/algebraic_topology/dold_kan/homotopies.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import algebra.homology.homotopy
+import algebraic_topology.dold_kan.notations
+
+/-!
+
+# Construction of homotopies for the Dold-Kan correspondence
+
+(The general strategy of proof of the Dold-Kan correspondence is explained
+in `equivalence.lean`.)
+
+The purpose of the files `homotopies.lean`, `faces.lean`, `projections.lean`
+and `p_infty.lean` is to construct an idempotent endomorphism
+`P_infty : K[X] ⟶ K[X]` of the alternating face map complex
+for each `X : simplicial_object C` when `C` is a preadditive category.
+In the case `C` is abelian, this `P_infty` shall be the projection on the
+normalized Moore subcomplex of `K[X]` associated to the decomposition of the
+complex `K[X]` as a direct sum of this normalized subcomplex and of the
+degenerate subcomplex.
+
+In `p_infty.lean`, this endomorphism `P_infty` shall be obtained by
+passing to the limit idempotent endomorphisms `P q` for all `(q : ℕ)`.
+These endomorphisms `P q` are defined by induction. The idea is to
+start from the identity endomorphism `P 0` of `K[X]` and to ensure by
+induction that the `q` higher face maps (except $d_0$) vanish on the
+image of `P q`. Then, in a certain degree `n`, the image of `P q` for
+a big enough `q` will be contained in the normalized subcomplex. This
+construction is done in `projections.lean`.
+
+It would be easy to define the `P q` degreewise (similarly as it is done
+in *Simplicial Homotopy Theory* by Goerrs-Jardine p. 149), but then we would
+have to prove that they are compatible with the differential (i.e. they
+are chain complex maps), and also that they are homotopic to the identity.
+These two verifications are quite technical. In order to reduce the number
+of such technical lemmas, the strategy that is followed here is to define
+a series of null homotopic maps `Hσ q` (attached to families of maps `hσ`)
+and use these in order to construct `P q` : the endomorphisms `P q`
+shall basically be obtained by altering the identity endomorphism by adding
+null homotopic maps, so that we get for free that they are morphisms
+of chain complexes and that they are homotopic to the identity. The most
+technical verifications that are needed about the null homotopic maps `Hσ`
+are obtained in `faces.lean`.
+
+In this file `homotopies.lean`, we define the null homotopic maps
+`Hσ q : K[X] ⟶ K[X]`, show that they are natural (see `nat_trans_Hσ`) and
+compatible the application of additive functors (see `map_Hσ`).
+
+## References
+* Albrecht Dold, Homology of Symmetric Products and Other Functors of Complexes,
+Annals of Mathematics, Second Series, Vol. 68 No. 1 (Jul. 1958), pp. 54-80.
+* Paul G. Goerss, John F. Jardine, Simplical Homotopy Theory, Modern Birkhäuser Classics,
+Reprint of the 1999 edition.
```

Comment:
Could you please add these to reference.bib?

Reply:
Thanks for the review! I am doing the changes.